packages/shylu/shylu_dd/frosch/src/CoarseSpaces/FROSch_CoarseSpace_def.hpp

```diff
@@ -176,12 +174,10 @@ namespace FROSch {
 
         LO iD;
         SC valueTmp;
-        GOVec indices;
-        SCVec values;
 
         for (UN i=0; i<AssembledBasis_->getLocalLength(); i++) {
-            indices.resize(0);
-            values.resize(0);
+            GOVec indices;
```

Comment:
The resize approach wasn't actually so bad.  `std::vector`, and therefore `Teuchos::Array`, do not necessarily free memory if you resize to zero.  Thus, resizing the existing vectors might have let you reuse storage.  I don't think that will affect performance much, though.